### DISCUSSION TITLE
Font display issues when publishing any two or more DWG files with Chinese fonts into a bim file

### DISCUSSION THREAD

**Discussion by ghost** [iTwin/itwinjs-core#6891] [General] [answered]
When using the snapshot tool to publish any two or more DWG files with Chinese fonts into a BIM file, there are font display issues. Only the content of the first DWG file is displayed correctly.

**These are the two DWG files I tested:**
[file.zip](https://github.com/user-attachments/files/15955841/file.zip)

![file1](https://github.com/iTwin/itwinjs-core/assets/34014227/34940c57-8817-4708-bf48-0ae0f12b0a08)
![file2](https://github.com/iTwin/itwinjs-core/assets/34014227/ee19c23c-d632-49ac-be96-855b3bb9df06)

**This is the result after publishing them into a BIM file:**

![snapshot_file1](https://github.com/iTwin/itwinjs-core/assets/34014227/d995118a-0f7b-4346-a8ee-58e91162fd5a)
![snapshot_file2](https://github.com/iTwin/itwinjs-core/assets/34014227/6c6c8a94-c3ce-48a8-a197-9c2c46c1e122)

**Comment by pmconne** [comment]
Thank you for providing the files.
Is this a duplicate of #6849?
@jffmarker @abeesh

**Reply by jffmarker** [reply]
Can you please try to reproduce @donhfu ?

**Reply by ghost** [reply]
> Thank you for providing the files. Is this a duplicate of #6849? @jffmarker @abeesh

No, I am not the same person as the author of https://github.com/iTwin/itwinjs-core/discussions/6849.
I think you can delete https://github.com/iTwin/itwinjs-core/discussions/6849 because my post explains it more clearly.

**Reply by donhfu** [reply]
Presumably the two models were created from two separate jobs from the two DWG files.  @guanhuaming, is it possible that the two files were from two different folders, and one folder contained font file FangSong_GB2312.ttf while the other did not?  If both folders have the font file, we'd need this file in hopes to reproduce what you have seen.

**Reply by ghost** [reply]
> Presumably the two models were created from two separate jobs from the two DWG files. @guanhuaming, is it possible that the two files were from two different folders, and one folder contained font file FangSong_GB2312.ttf while the other did not? If both folders have the font file, we'd need this file in hopes to reproduce what you have seen.

I placed the FangSong_GB2312.ttf font in the font directory of RealDwgBridge, instead of placing it in the same directory as the DWG files. The two DWG files are located in the same directory.

If I place the font in the RealDwgBridge font directory, should all DWG files share this font during conversion?

The font directory of RealDwgBridge is as follows:
`C:\Users\Administrator\AppData\Local\imodelbridges\RealDwgBridge\Fonts`

By the way, the FangSong_GB2312.ttf font is a standard system font in Windows 7. I am currently using Windows 11.
FangSong_GB2312.ttf font download:
[FangSong_GB2312.zip](https://github.com/user-attachments/files/15958647/FangSong_GB2312.zip)

**Reply by donhfu** [reply]
Since you are running a local connector, you have done it correctly placing fonts under \Fonts\ sub-folder.  I have trouble reproducing the inconsistent display from the two files when placing your font under \Fonts\.  They both are displayed correctly for me.  But my box is in English locale which might be different than yours.  We will try some more tests and see if we can reproduce the incorrect display.

**Reply by donhfu** [reply]
@guanhuaming, please try opening and editing file \Assets\ConvertConfig.xml, find the font mapping section in middle of the file, and add a new entry:
```
    <TrueTypeFont TypeFace="仿宋_GB2312" AltTypeFace="FangSong_GB2312" AltShxName=""/>
```
This will get around the mismatched family name for the true type font.  We will get the root problem fixed in a future connector release.  When the release will get deployed, you will no longer have to manually map your font (although you still can keep the mapping as you wish).  Please give this a shot and see if it works for you.  Thanks for brining this issue to our attention!

**Reply by ghost** [reply, marked as answer]
> This will get around the mismatched family name for the true type font. We will get the root problem fixed in a future connector release. When the release will get deployed, you will no longer have to manually map your font (although you still can keep the mapping as you wish). Please give this a shot and see if it works for you. Thanks for brining this issue to our attention!

Tested and it works for me.  Thank you very much.
